packages/core/useWindowScroll/index.ts

```diff
@@ -1,30 +1,85 @@
-import { ref } from 'vue-demi'
+import type { MaybeComputedRef } from '@vueuse/shared'
+import { resolveRef, resolveUnref } from '@vueuse/shared'
+import type { Ref } from 'vue-demi'
+import { computed, ref } from 'vue-demi'
 import { useEventListener } from '../useEventListener'
 import type { ConfigurableWindow } from '../_configurable'
 import { defaultWindow } from '../_configurable'
 
+export interface UseWindowScrollOptions extends ConfigurableWindow {
+  /**
+   * Optionally specify a scroll behavior of `auto` (default, not smooth scrolling) or `smooth` (for smooth scrolling)
+   *
+   * @default 'auto'
+   */
+  behavior?: MaybeComputedRef<ScrollBehavior>
+}
+
+export interface UseWindowScrollReturn {
+  /**
+   * The X scroll position of the page
+   */
+  x: Ref<number>
+  /**
+   * The Y scroll position of the page
+   */
+  y: Ref<number>
+}
+
 /**
  * Reactive window scroll.
  *
  * @see https://vueuse.org/useWindowScroll
  * @param options
  */
-export function useWindowScroll({ window = defaultWindow }: ConfigurableWindow = {}) {
-  if (!window) {
+export function useWindowScroll(options: UseWindowScrollOptions = {}): UseWindowScrollReturn {
+  const window = resolveRef(options.window || defaultWindow)
```

Comment:
Is `window` needs to be a ref?

Reply:
Good catch, it does not because the TS type wasn't saying it was a `MaybeComputedRef` anyway. I removed the `unref`: https://github.com/vueuse/vueuse/compare/71821f5a1d21e2e77378093988c5d95e775b4692..dbeb578b30f5899d201bc136e4658cc9b2793033